### PR TITLE
Add & validate more fields for plugin viewer

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Adds more fields to plugin's YAML

| Property | YAML  |
|---|---|
| Title | title |
| Publisher | publisher |
| Category | category |
| Icon | icon |
| Short-Description | description |
| Preview | preview | 
| Repository | repository
| First Publication Date | firstPublicationDate | 
| Latest Updated | latestUpdateDate |
| Tags | tags
| Media-img | media-img
| Media-video | media-video

Added script to validate presense of mandatory fields (to not be null or empty):
`title, publisher, category, icon, description, repository, firstPublicationDate, latestUpdateDate`

Field `category` is also validated, as it must have the value of one of the following 
"Editor", "Debugger", "Formatter", "Language", "Linter", "Snippet", "Theme", "Other"

Non mandatory fields are not validated at all

### What issue does this PR fix or reference
https://github.com/eclipse/che-plugin-registry/issues/71

### CODE REVIEW

File: check_plugin_viewer_fields.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (benoitf) [inline check_plugin_viewer_fields.sh]: hello, could you please validate on https://www.shellcheck.net/ (there are lot of warnings)

File: plugins/che-dummy-plugin/0.0.1/meta.yaml [hunk 2]
```diff
@@ -6,3 +6,12 @@ title: Che Samples Hello World Plugin
 description: A hello world theia plug-in wrapped into a Che Plug-in
 icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
 url: https://github.com/ws-skeleton/che-dummy-plugin/releases/download/untagged-8f3e198285a2f3b6b2db/che-dummy-plugin.tar.gz
+publisher: Red Hat, Inc.
+category: Other
+repository: https://github.com/ws-skeleton/che-dummy-plugin/
+firstPublicationDate: Jan 11, 2019
+latestUpdateDate: Jan 11, 2019
+preview:
```
Comment (benoitf) [inline plugins/che-dummy-plugin/0.0.1/meta.yaml]: if fields are empty, why are they mandatory ?

File: plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml [hunk 3]
```diff
@@ -6,3 +6,12 @@ title: Jupyter Notebook as Editor for Eclipse Che
 description: Jupyter Notebook as Editor for Eclipse Che
 icon: https://jupyter.org/assets/main-logo.svg
 url: https://github.com/ws-skeleton/che-editor-jupyter/releases/download/untagged-cc4e4c9a7741551b6776/che-editor-plugin.tar.gz
+publisher: Red Hat, Inc.
+category: Editor
+repository: https://github.com/ws-skeleton/che-editor-jupyter/
+firstPublicationDate: Jan 11, 2019
```
Comment (benoitf) [inline plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml]: what is the format of this date ?

Comment (skabashnyuk) [reply plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml]: We have no inputs about that. What you can suggest?

File: plugins/org.eclipse.che.editor.gwt/1.0.0/meta.yaml [hunk 4]
```diff
@@ -6,3 +6,12 @@ title: Eclipse GWT IDE for Eclipse Che
 description: Eclipse GWT IDE
 icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
 url: https://github.com/eclipse/che-editor-gwt-ide/releases/download/che-editor-gwt-ide-2018-12-18_2212-00/che-editor-plugin.tar.gz
+publisher: Red Hat, Inc.
+category: Editor
+repository: https://github.com/eclipse/che-editor-gwt-ide/
+firstPublicationDate: Jan 11, 2019
+latestUpdateDate: Jan 11, 2019
```
Comment (benoitf) [inline plugins/org.eclipse.che.editor.gwt/1.0.0/meta.yaml]: this field could be also automatically generated when building the image

Comment (benoitf) [reply plugins/org.eclipse.che.editor.gwt/1.0.0/meta.yaml]: I mean, the field is important when we read data not when we create yaml file.

AFAIK the script could add the fields when building the image if missing

File: plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml (hunk 3 above)
Comment (benoitf) [reply plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml]: https://en.wikipedia.org/wiki/ISO_8601 ?

File: plugins/che-dummy-plugin/0.0.1/meta.yaml (hunk 2 above)
Comment (mkuznyetsov) [reply plugins/che-dummy-plugin/0.0.1/meta.yaml]: they're not, and I shall remove them

File: plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml (hunk 3 above)
Comment (mkuznyetsov) [reply plugins/org.eclipse.che.editor.jupyter/1.0.0/meta.yaml]: ok, I'll use this format

File: check_plugin_viewer_fields.sh (hunk 1 above)
Comment (mkuznyetsov) [reply check_plugin_viewer_fields.sh]: thx, I addressed the warnings in it

File: Dockerfile [hunk 5]
```diff
@@ -1,14 +1,13 @@
-FROM mikefarah/yq
+FROM mikefarah/yq as builder
```
Comment (benoitf) [inline Dockerfile]: We could add copyright headers there (they were not there before your changes so..)

File: plugins/che-dummy-plugin/0.0.1/meta.yaml [hunk 6]
```diff
@@ -6,3 +6,6 @@ title: Che Samples Hello World Plugin
 description: A hello world theia plug-in wrapped into a Che Plug-in
 icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
 url: https://github.com/ws-skeleton/che-dummy-plugin/releases/download/untagged-8f3e198285a2f3b6b2db/che-dummy-plugin.tar.gz
+publisher: Red Hat, Inc.
+category: Other
```
Comment (benoitf) [inline plugins/che-dummy-plugin/0.0.1/meta.yaml]: what is the default category if there is one ? 
Could we assume that if no category is set, it goes to Other ?

File: set_plugin_dates.sh [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    cd "${VERSION_DIR_NAME}" || exit 1
+
+    DATE=$(date -I)
+    yq w meta.yaml firstPublicationDate "${DATE}" -i || exit 1
+    yq w meta.yaml latestUpdateDate "${DATE}" -i || exit 1
+    cd ..
+  done
+
+  cd ..
+done
```
Comment (benoitf) [inline set_plugin_dates.sh]: missing the EOF

**Comment by mkuznyetsov** [comment]
Updated the Dockerfiels and build scripts to accomodate for multi-staged build

File: Dockerfile (hunk 5 above)
Comment (mkuznyetsov) [reply Dockerfile]: ok

File: plugins/che-dummy-plugin/0.0.1/meta.yaml (hunk 6 above)
Comment (mkuznyetsov) [reply plugins/che-dummy-plugin/0.0.1/meta.yaml]: ok

File: set_plugin_dates.sh (hunk 7 above)
Comment (mkuznyetsov) [reply set_plugin_dates.sh]: will fix

**Comment by skabashnyuk** [comment]
@benoitf are you ok with current state of pr?

**Pull request merged**